File: in_Java/1394 Find Lucky Integer in an Array.java
class Solution {
    public int findLucky(int[] arr) {
        Map<Integer,Integer> hm = new HashMap<>();
        for(int i : arr){
            if(!hm.containsKey(i)) hm.put(i,1);
            else hm.put(i,hm.get(i)+1);
        }
        int res = Integer.MIN_VALUE;
        for(int k : hm.keySet()){
            if(k == hm.get(k)){
                res = res < k? k : res;
            }
        }
        return res < 0? -1 : res;
    }
}

File: in_Python/1394 Find Lucky Integer in an Array.py
class Solution(object):
    def findLucky(self, arr):
        """
        :type arr: List[int]
        :rtype: int
        """
        res = -1
        dict = {}
        for i in arr:
            if i not in dict:
                dict[i] = 1
            else:
                dict[i] += 1
        for i in dict:
            if i == dict[i]:
                res = max(res,i)
        return res